contracts/router/src/contract/execute.rs

```diff
@@ -160,11 +223,21 @@ fn validate_msgs(
 }
 
 pub fn route_messages(
-    store: impl Store,
+    mut store: impl Store,
     sender: Addr,
     msgs: Vec<Message>,
 ) -> error_stack::Result<Response, Error> {
-    let config = store.load_config()?;
+    STATE
+        .load(store.storage())
+        .change_context(Error::StoreFailure)
+        .map(|state| match state {
+            State::Disabled => Err(Error::RoutingDisabled),
+            State::Enabled => Ok(()),
+        })??;
```

Comment:
```suggestion
     if STATE
        .load(store.storage())
        .change_context(Error::StoreFailure)? == State::Disabled {
            return Err(report!(Error::RoutingDisabled));
        }
```

I find this easier to read